File: C++/SpecHandphone.cpp
#include <iostream>
#include <string>
using namespace std;

class SpecHandphone : public Produk {
private:
    string processor;
    int ram;
    int storage;
    int battery;

public:
    // Setter
    void setProcessor(const string &p) { processor = p; }
    void setRAM(int r) { ram = r; }
    void setStorage(int s) { storage = s; }
    void setBattery(int b) { battery = b; }

    // Getter
    string getProcessor() const { return processor; }
    int getRAM() const { return ram; }
    int getStorage() const { return storage; }
    int getBattery() const { return battery; }
};

File: PHP/SpecHandphone.php
<?php
require_once "Produk.php";

class SpecHandphone extends Produk {
    private $processor;
    private $ram;
    private $storage;
    private $battery;

    public function __construct($id_produk = "", $nama = "", $harga = 0, $image = "",
                                $processor = "", $ram = 0, $storage = 0, $battery = 0) {
        parent::__construct($id_produk, $nama, $harga, $image);
        $this->processor = $processor;
        $this->ram = $ram;
        $this->storage = $storage;
        $this->battery = $battery;
    }

    public function getProcessor() { return $this->processor; }
    public function setProcessor($processor) { $this->processor = $processor; }

    public function getRam() { return $this->ram; }
    public function setRam($ram) { $this->ram = $ram; }

    public function getStorage() { return $this->storage; }
    public function setStorage($storage) { $this->storage = $storage; }

    public function getBattery() { return $this->battery; }
    public function setBattery($battery) { $this->battery = $battery; }
}
?>
